library/kani/src/shadow.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+extern "C" {
+    pub fn __KANI_pointer_object(ptr: *const u8) -> usize;
```

Comment:
I would prefer not adding a new way of implementing Kani intrinsic for now. Can we just use a hook to keep it consistent with other Kani operations?

Reply:
Sure.

Reply:
Done.